dartsim/src/SDFFeatures.cc

```diff
@@ -342,59 +372,82 @@ Identity SDFFeatures::ConstructSdfWorld(
   // information here. For now, we'll just use dartsim's default physics
   // parameters.
 
-  for (std::size_t i=0; i < _sdfWorld.ModelCount(); ++i)
+  for (std::size_t i = 0; i < _sdfWorld.ModelCount(); ++i)
   {
     const ::sdf::Model *model = _sdfWorld.ModelByIndex(i);
 
     if (!model)
       continue;
 
-    this->ConstructSdfModel(worldID, *model);
+    this->ConstructSdfNestedModel(worldID, *model);
   }
 
   return worldID;
 }
 
 /////////////////////////////////////////////////
 Identity SDFFeatures::ConstructSdfModel(
-    const Identity &_worldID,
+    const Identity &_parentID,
     const ::sdf::Model &_sdfModel)
 {
-  // check if parent is a world
-  if (!this->worlds.HasEntity(_worldID))
+  return this->ConstructSdfModelImpl(_parentID, _sdfModel, "");
+}
+
+/////////////////////////////////////////////////
+Identity SDFFeatures::ConstructSdfNestedModel(const Identity &_parentID,
+                                              const ::sdf::Model &_sdfModel)
+{
+  return this->ConstructSdfModelImpl(_parentID, _sdfModel, "");
+}
+
+/////////////////////////////////////////////////
+Identity SDFFeatures::ConstructSdfModelImpl(
+    std::size_t _parentID,
+    const ::sdf::Model &_sdfModel, const std::string &)
+{
+  auto worldID = _parentID;
+  std::string modelName = _sdfModel.Name();
+  // If this is a nested model, find the world assocated with the model
+  if (this->models.HasEntity(_parentID))
   {
-    ignerr << "Unable to construct model: " << _sdfModel.Name() << ". "
-           << "Parent of model is not a world. " << std::endl;
-    return this->GenerateInvalidId();
+    worldID = this->models.idToContainerID.at(_parentID);
+    const auto &skel = this->models.at(_parentID)->model;
+    modelName = ::sdf::JoinName(skel->getName(), _sdfModel.Name());
   }
 
+  dart::dynamics::Frame *parentFrame = this->frames.at(_parentID);
+
   dart::dynamics::SkeletonPtr model =
-      dart::dynamics::Skeleton::create(_sdfModel.Name());
+      dart::dynamics::Skeleton::create(modelName);
 
   dart::dynamics::SimpleFramePtr modelFrame =
       dart::dynamics::SimpleFrame::createShared(
-        dart::dynamics::Frame::World(),
-        _sdfModel.Name()+"_frame",
-        ResolveSdfPose(_sdfModel.SemanticPose()));
+          parentFrame, modelName + "::__model__",
+          ResolveSdfPose(_sdfModel.SemanticPose()));
 
   // Set canonical link name
   auto [modelID, modelInfo] = this->AddModel( // NOLINT
-      {model, modelFrame, _sdfModel.CanonicalLinkName()}, _worldID);
+      {model, modelFrame, _sdfModel.CanonicalLinkName()}, worldID);
```

Comment:
In `ConstructSdfLink`, this canonical link name is associated with a link. I would think we need special handling of that now that the link can be nested.

Reply:
After meeting we decided this feature addition would be left as a todo. I opened an issue here for tracking

https://github.com/ignitionrobotics/ign-physics/issues/209